core/client/hdfs/src/main/java/alluxio/hadoop/FileSystem.java

```diff
@@ -94,6 +97,33 @@ protected boolean isZookeeperMode() {
       // Unset the zookeeper configuration to support alluxio URI has the highest priority
       alluxioConfProperties.put(PropertyKey.ZOOKEEPER_ENABLED.getName(), false);
       alluxioConfProperties.put(PropertyKey.ZOOKEEPER_ADDRESS.getName(), null);
+    } else if (alluxioUri.getAuthority() instanceof EmbeddedLogicalAuthority) {
+      EmbeddedLogicalAuthority authority = (EmbeddedLogicalAuthority) alluxioUri.getAuthority();
+      String confName = PropertyKey.MASTER_RPC_ADDRESSES.getName()
+          + "." + authority.getLogicalName();
+      if (conf.get(confName) != null) {
+        // If user set alluxio.master.rpc.addresses.[logicalName]
+        String masterAddress = conf.get(confName);
+        alluxioConfProperties.put(PropertyKey.MASTER_RPC_ADDRESSES.getName(), masterAddress);
+      } else {
+        throw new IllegalStateException(
+            String.format("Invalid uri. You must set %s to use the logical name ", confName));
+      }
+      alluxioConfProperties.put(PropertyKey.ZOOKEEPER_ENABLED.getName(), false);
+      alluxioConfProperties.put(PropertyKey.ZOOKEEPER_ADDRESS.getName(), null);
+    } else if (alluxioUri.getAuthority() instanceof ZookeeperLogicalAuthority) {
+      ZookeeperLogicalAuthority authority = (ZookeeperLogicalAuthority) alluxioUri.getAuthority();
+      String confName =
+          PropertyKey.ZOOKEEPER_ADDRESS.getName() + "." + authority.getLogicalName();
```

Comment:
Where is this configuration defined?

Reply:
Is this configuration provided in your private environment?

Reply:
I've written the documentation and can explain the issue.

Reply:
Thanks for adding some documentation! If possible please hold off on this PR until we find the causes of #13967 (and #14019). We are looking into them.

Reply:
@ggezer  I understand what you mean and have added the new configuration to `PropertyKey.java`.